chia/cmds/init_funcs.py

```diff
@@ -236,32 +241,53 @@ def create_all_ssl(root_path: Path):
     chia_ca_key_path = ca_dir / "chia_ca.key"
     write_ssl_cert_and_key(chia_ca_crt_path, chia_ca_crt, chia_ca_key_path, chia_ca_key)
 
+    # If Private CA crt/key are passed-in, write them out
+    if private_ca_crt_and_key is not None:
+        private_ca_crt, private_ca_key = private_ca_crt_and_key
+        write_ssl_cert_and_key(private_ca_crt_path, private_ca_crt, private_ca_key_path, private_ca_key)
+
     if not private_ca_key_path.exists() or not private_ca_crt_path.exists():
         # Create private CA
         print(f"Can't find private CA, creating a new one in {root_path} to generate TLS certificates")
         make_ca_cert(private_ca_crt_path, private_ca_key_path)
         # Create private certs for each node
         ca_key = private_ca_key_path.read_bytes()
         ca_crt = private_ca_crt_path.read_bytes()
-        generate_ssl_for_nodes(ssl_dir, ca_crt, ca_key, True)
+        generate_ssl_for_nodes(
+            ssl_dir, ca_crt, ca_key, nodes=private_node_names, private=True, node_certs_and_keys=node_certs_and_keys
+        )
     else:
         # This is entered when user copied over private CA
         print(f"Found private CA in {root_path}, using it to generate TLS certificates")
         ca_key = private_ca_key_path.read_bytes()
         ca_crt = private_ca_crt_path.read_bytes()
-        generate_ssl_for_nodes(ssl_dir, ca_crt, ca_key, True)
+        generate_ssl_for_nodes(
+            ssl_dir, ca_crt, ca_key, nodes=private_node_names, private=True, node_certs_and_keys=node_certs_and_keys
+        )
 
     chia_ca_crt, chia_ca_key = get_chia_ca_crt_key()
-    generate_ssl_for_nodes(ssl_dir, chia_ca_crt, chia_ca_key, False, overwrite=False)
-
-
-def generate_ssl_for_nodes(ssl_dir: Path, ca_crt: bytes, ca_key: bytes, private: bool, overwrite=True):
-    if private:
-        names = private_node_names
-    else:
-        names = public_node_names
-
-    for node_name in names:
+    generate_ssl_for_nodes(
+        ssl_dir,
+        chia_ca_crt,
+        chia_ca_key,
+        nodes=public_node_names,
+        private=False,
+        overwrite=False,
+        node_certs_and_keys=node_certs_and_keys,
+    )
+
+
+def generate_ssl_for_nodes(
+    ssl_dir: Path,
+    ca_crt: bytes,
+    ca_key: bytes,
+    *,
+    nodes: List[str],
+    private: bool,
+    overwrite: bool = True,
+    node_certs_and_keys: Optional[Dict[str, Dict]] = None,
+):
+    for node_name in nodes:
```

Comment:
previously we would use `private` to pick which list we would iterate, but now we pass in that list instead. The need for the `private` parameter is not diminished to just affecting the prefix (and it seems like it might be clearer to just pass in the prefix instead, and have it default to "public".

The new API of this function allows you to make the prefix "private" but still use the public nodes, and vice versa. If we at least passed in the prefix instead of the `bool`, I think it would be a bit clearer at the call site if we mix and match public and private.

Reply:
Updated. The `private` param is now named `prefix` and takes a string value.